hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/S3GatewayConfigKeys.java

```diff
@@ -63,7 +63,7 @@ public final class S3GatewayConfigKeys {
       "ozone.s3g.kerberos.keytab.file";
   public static final String OZONE_S3G_KERBEROS_PRINCIPAL_KEY =
       "ozone.s3g.kerberos.principal";
-
+  public static final boolean OZONE_S3G_FSO_DIRECTORY_CREATION_ENABLED = true;
```

Comment:
This also needs a constant for the config key the user would actually add. Like `OZONE_S3G_KERBEROS_PRINCIPAL_KEY` above, for example.

Reply:
Also let's put a block comment here to explain what this config means. I filed HDDS-7824 for a more widespread user facing doc update.